test/protocols/_waitfor.cxx

```diff
@@ -0,0 +1,74 @@
+/********************************************************************************
+ * Copyright (C) 2014-2018 GSI Helmholtzzentrum fuer Schwerionenforschung GmbH  *
+ *                                                                              *
+ *              This software is distributed under the terms of the             *
+ *              GNU Lesser General Public Licence (LGPL) version 3,             *
+ *                  copied verbatim in the file "LICENSE"                       *
+ ********************************************************************************/
+
+#include "runner.h"
+
+#include <gtest/gtest.h>
+#include <boost/process.hpp>
+
+#include <sys/types.h>
+#include <signal.h>
+
+#include <string>
+#include <thread>
+#include <chrono>
+#include <iostream>
+#include <future> // std::async, std::future
+
+namespace
+{
+
+using namespace std;
+using namespace fair::mq::test;
+
+void RunWaitFor()
+{
+    std::mutex mtx;
+    std::condition_variable cv;
+
+    int pid = 0;
+    int exit_code = 0;
+
+    thread deviceThread([&]() {
+        stringstream cmd;
+        cmd << runTestDevice << " --id waitfor_" << " --control static " << " --severity nolog";
+
+        boost::process::ipstream stdout;
+        boost::process::child c(cmd.str(), boost::process::std_out > stdout);
+        string line;
+        getline(stdout, line);
+
+        {
+            std::lock_guard<std::mutex> lock(mtx);
+            pid = std::stoi(line);
```

Comment:
I think you can get the child process id with `c.id()` here.

Reply:
seems that i missed the id method :) will try tomorrow

Reply:
hmm, but then i will need another sleep. if i just take id from boost, and kill it right away, i will likely kill it before waitfor starts. although there is still some chance of bad timing now (string evaluated and converted before second process proceeds), but much less likely

Reply:
Changed to using `id()` and removed pid stuff from device.